File: 01-Front/Saniteau/client-src/model/abonne/Abonne.ts
import { Tarification } from '../../model/abonne/Tarification';


export class Abonne {
    public idAbonne: number;
    public idAdresse: number;
    public nom: string;
    public prenom: string;
    public tarification: Tarification;
    public numeroEtRue: string;
    public ville :string;
    public codePostal :string;


    constructor(idAbonne: number, idAdresse: number, nom: string, prenom: string, tarification: Tarification, numeroEtRue: string, ville: string, codePostal: string ) {
        this.idAbonne = idAbonne;
        this.idAdresse = idAdresse;
        this.nom = nom;
        this.prenom = prenom;
        this.tarification = tarification;
        this.numeroEtRue = numeroEtRue;
        this.ville = ville;
        this.codePostal = codePostal;
    }

}

File: 01-Front/Saniteau/Models/Abonnés/Abonne.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Saniteau.Models.Abonnés
{
    public class Abonne
    {
        public int IdAbonne { get; set; }
        public int IdAdresse { get; set; }
        public string? Nom { get; set; }
        public string? Prenom { get; set; }
        public string? NumeroEtRue { get; set; }
        public string? Ville { get; set; }
        public string? CodePostal { get; set; }
        public Tarification Tarification  { get; set; }

        public Abonne()
        {

        }

        public Abonne(int idAbonne , int idAdresse, string? nom, string? prenom, Tarification tarification, string? numeroEtRue, string? ville, string? codePostal)
        {
            IdAbonne = idAbonne;
            IdAdresse = idAdresse;
            Nom = nom;
            Prenom = prenom;
            Tarification = tarification;
            NumeroEtRue = numeroEtRue;
            Ville = ville;
            CodePostal = codePostal;
        }

    }
}
